decode.go

```diff
@@ -468,10 +483,7 @@ func (d *decoder) value(v interface{}) error {
 		return &InvalidUnmarshalError{"cbor: Unmarshal(nil " + rv.Type().String() + ")"}
 	}
 
-	off := d.off // Save offset before data validation
-	err := d.valid()
-	d.off = off // Restore offset
-	if err != nil {
+	if err := d.Valid(); err != nil {
```

Comment:
By saving and restoring data offset in the calling function `decoder.value()`, `decoder.valid()` can validate multiple CBOR data instead of just the first one.  

Would it affect your project if we keep the code of saving and restoring offset in `decoder.value()`?

Reply:
That makes sense, I reverted in 4f2812f